### ISSUE TITLE
Document steps for contributors working on new items

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#172]
I propose the following steps be followed for each diagnostic:

> 1. To start working on a diagnostic, add a link to the page containing the reference documentation for the warning and indicate you are currently working on implementing it.
> 2. Add a new issue for a code fix for the diagnostic. For example, I added #171 when I worked on #6. Even if no code fix is possible, the issue is a place for discussions regarding possible corrections. Code fixes may, but do not have to be implemented alongside the diagnostic.
> 3. If a diagnostic or code fix is submitted without tests, it might be rejected. However, it may be accepted provided all of the following are true:
>    1. The code is disabled by default. It will be enabled by default only after tests are in place.
>    2. A new issue was created for implementing tests for the item.
>    3. Evidence was given that the feature is currently operational, and the code appears to be a solid starting point for other contributors to continue the implementation effort.

Following discussion of this topic, the instructions should be added to a **CONTRIBUTING.md** file in the project root.
